# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1216416810 -7200
#      Fri Jul 18 23:33:30 2008 +0200
# Node ID efabe34cce70d47d55b629725051acf47cb16208
# Parent  2c0a0edae59664192af244af532bab5db1b74254
make configure compatible with FTGL ver <= 2.1.2

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2008-07-21  Jaroslav Hajek <highegg@gmail.com>
+
+	* configure.in: Extend FTGL test to test for both ftgl.h and FTGL.h.
+	Use HAVE_FTGL_UPPERCASE to indicate the latter (version <= 2.1.2).
+
 2008-07-19 Rafael Laboissiere <rafael@debia8n.org>
 
 	* octMakefile.in (CONF_DISTFILES): Include acx_blas_f77_func.m4 in
 	the list.
 
 2008-07-18 Carlo de Falco <carlo.defalco@gmail.com>
 
 	* aclocal.m4: Search for gl.h and glu.h in OpenGL/ as well as in GL/.
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -659,29 +659,41 @@ if test -n "$OPENGL_LIBS"; then
   GRAPHICS_OPENGL="opengl"
   GL_RENDER_SRC="gl-render.cc"
 
 ## ftgl (needs freetype 2)
   AC_CHECK_FT2([9.0.3],[],
     [warn_freetype="FreeType library not found. Native renderer will not have on-screen text"])
   if test -z "$warn_freetype"; then
     AC_LANG_PUSH(C++)
-      AC_CHECK_HEADERS([FTGL/ftgl.h ftgl.h])
-      if test "$ac_cv_header_FTGL_ftgl_h" = yes \
-        || test "$ac_cv_header_ftgl_h" = yes; then
+      has_ftgl_h=yes
+      AC_CHECK_HEADERS([FTGL/ftgl.h ftgl.h], break, [
+	AC_CHECK_HEADERS([FTGL/FTGL.h FTGL.h], [
+          AC_DEFINE(HAVE_FTGL_UPPERCASE,1,[Define to 1 if you have FTGL.h or FTGL/FTGL.h])
+	  break
+	  ], has_ftgl_h=no)])
+      if test "$has_ftgl_h" = yes; then
 	AC_MSG_CHECKING([for FTGLTextureFont in -lftgl])
         save_LIBS="$LIBS"
 	save_CXXFLAGS="$CXXFLAGS"
 	LIBS="$LIBS $OPENGL_LIBS $FT2_LIBS -lftgl"
 	CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
         AC_TRY_LINK([
 #ifdef HAVE_FTGL_FTGL_H
+#ifdef HAVE_FTGL_UPPERCASE
+#include <FTGL/FTGL.h>
+#else
 #include <FTGL/ftgl.h>
+#endif
 #elif HAVE_FTGL_H
+#ifdef HAVE_FTGL_UPPERCASE
+#include <FTGL.h>
+#else
 #include <ftgl.h>
+#endif
 #endif], [
 FTGLTextureFont font("");], [
           OPENGL_LIBS="$OPENGL_LIBS $FT2_LIBS -lftgl"
 	  LIBS="$save_LIBS"
           AC_DEFINE(HAVE_FTGL, 1, [Define to 1 if FTGL is present])
 	  AC_MSG_RESULT(yes)
 	  XTRA_CXXFLAGS="$XTRA_CXXFLAGS $FT2_CFLAGS"],[
 	  AC_MSG_RESULT(no)
